File: backend/app/models/project.py
# Project Model
# -------------
# Fields: id, name, description, key (unique slug),
#         status (enum: ACTIVE/ARCHIVED/ON_HOLD),
#         owner_id (FK → users), start_date, end_date,
#         created_at, updated_at
#
# Relationships: sprints, tasks, members (M2M → users),
#                scorecards
#
# Placeholder — implementation pending.

File: frontend/src/types/project.ts
/* Project Types */

export type ProjectStatus = "ACTIVE" | "ARCHIVED";

export interface Project {
  id: string;
  name: string;
  description: string | null;
  status: ProjectStatus;
  owner_id: string;
  created_at: string;
  updated_at: string;
}

export interface ProjectSummary {
  id: string;
  name: string;
  status: ProjectStatus;
}

export interface ProjectMember {
  user_id: string;
  role: "owner" | "admin" | "member" | "viewer";
}
